flake.nix

```diff
@@ -0,0 +1,108 @@
+{
+  description = "slugger: Clean URI slugs for Haskell";
+
+  nixConfig.bash-prompt = "[nix]λ ";
+
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixos-21.05";
+
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { flake-utils, nixpkgs, self }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        config = {};
+
+        compilerVersion = "884";
+        compiler = "ghc" + compilerVersion;
+        overlays = [
+          (final: prev: {
+            haskell-language-server = prev.haskell-language-server.override {
+              supportedGhcVersions = [ compilerVersion ];
+            };
+
+            myHaskellPackages = prev.haskell.packages.${compiler}.override {
+              overrides = hpFinal: hpPrev: {
+                slugger = hpPrev.callCabal2nix "slugger" ./. {};
+
+                hspec = hpPrev.callHackage "hspec" "2.8.1" {}; # see slugger.cabal build-depends
+                hspec-core = hpPrev.callHackage "hspec-core" "2.8.1" {}; # because hspec
+                hspec-discover = hpPrev.callHackage "hspec-discover" "2.8.1" {}; # because hspec
+              };
+            };
+          })
+        ];
+
+        pkgs = import nixpkgs { inherit config overlays system; };
+      in rec {
+        packages = with pkgs.myHaskellPackages; [ slugger ];
+
+        devShell = pkgs.myHaskellPackages.shellFor {
+          packages = p: [
+            p.slugger
+          ];
+
+          buildInputs = with pkgs.myHaskellPackages; [
+            slugger
+
+            cabal-install
+            hlint                        # https://github.com/ndmitchell/hlint
+            ormolu                       # https://github.com/tweag/ormolu
+            pkgs.haskell-language-server # https://github.com/haskell/haskell-language-server
```

Comment:
To be honest I am a bit surprised that this works, my impression was that you had to use `haskell-language-server` from the same compiler (without `pkgs` here). But I don't know it very well, so I might be wrong.

Reply:
I passed a `compilerVersion` to HLS, and I'm building to `myHaskellPackages = prev.haskell.packages.${compiler}.override`, and I believe this works for me just fine.

Reply:
Oh, I see, thanks. To be honest I feel like just putting `haskell-langauge-server` there is simpler; but hey, this works. :)

Reply:
I believe HLS exists as a top-level package, and my concern was there would then be `myHaskellPackages.haskell-language-server` (the old one), as well as one defined at the compiler level